frappe/core/doctype/document_key/document_key.json

```diff
@@ -0,0 +1,74 @@
+{
+ "actions": [],
+ "allow_rename": 1,
+ "autoname": "format:DOCUMENT-KEY-{########}",
+ "creation": "2022-01-14 13:40:49.487646",
+ "doctype": "DocType",
+ "document_key_expiry": 90,
+ "editable_grid": 1,
+ "engine": "InnoDB",
+ "field_order": [
+  "reference_doctype",
+  "reference_docname",
+  "key",
+  "expires_on"
+ ],
+ "fields": [
+  {
+   "fieldname": "reference_doctype",
+   "fieldtype": "Link",
+   "label": "Reference Document Type",
+   "options": "DocType",
+   "read_only": 1,
+   "search_index": 1
+  },
+  {
+   "fieldname": "reference_docname",
+   "fieldtype": "Dynamic Link",
+   "label": "Reference Document Name",
+   "options": "reference_doctype",
+   "read_only": 1,
+   "search_index": 1
+  },
+  {
+   "fieldname": "key",
+   "fieldtype": "Data",
+   "label": "Key",
+   "read_only": 1
+  },
+  {
+   "fieldname": "expires_on",
+   "fieldtype": "Date",
+   "in_list_view": 1,
+   "label": "Expires On",
+   "read_only": 1
+  }
+ ],
+ "in_create": 1,
+ "index_web_pages_for_search": 1,
+ "links": [],
+ "modified": "2022-01-14 13:57:28.050678",
+ "modified_by": "Administrator",
+ "module": "Core",
+ "name": "Document Key",
```

Comment:
Should be `Document Share Key`?

Reply:
or "Document Print Key"... since it is currently used specifically for print-view?

--
Also, that dropdown of "Visibility" seems useless in our case. Do we need the ability to generate a key only for logged-in users (this was my initial thought for adding that dropdown)?

Reply:
- Visibility: not useful (confirmed by checking reference implementation)
- I think we need a way to distinguish this and existing Share feature. Maybe **Generate Link**? Although, **Share Link** has a nice ring to it. A totally different approach would be to show a link icon in below marked area:

![Screenshot-2022-01-20-203255](https://user-images.githubusercontent.com/16315650/150364737-589946f4-b4c6-406a-97ee-4b251f55293a.png)

This is somewhat similar to what Google does:

![image](https://user-images.githubusercontent.com/16315650/150365376-a18e1d5f-951f-4bf6-b408-399c22b5bcc8.png)

- I like **Document Share Key** better. Although it is technically a print view, print isn't the function.